dev/com.ibm.ws.security.wim.adapter.ldap_fat/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/LoginPropertyTest.java

```diff
@@ -0,0 +1,290 @@
+     * Configure the embedded LDAP server.
+     *
+     * @throws Exception If the server failed to start for some reason.
+     */
+    private static void setupldapServer() throws Exception {
+        ds = new InMemoryLDAPServer(DN);
+
+        Entry entry = new Entry(DN);
+        entry.addAttribute("objectclass", "top");
+        entry.addAttribute("objectclass", "domain");
+        ds.add(entry);
+
+        /*
+         * Create the user and group.
+         */
+        entry = new Entry(USER_DN);
+        entry.addAttribute("objectclass", "inetorgperson");
+        entry.addAttribute("uid", userUid);
+        entry.addAttribute("sn", userSn);
+        entry.addAttribute("cn", userCn);
+        entry.addAttribute("userPassword", "password");
+        ds.add(entry);
+
+    }
+
+    /**
+     * Convenience method to configure the Liberty server with an {@link LdapRegistry} configuration that
+     * will connect to {@link #ldapServer}.
+     *
+     * @throws Exception If there was an error configuring the server.
+     */
+    private static void updateLibertyServer(boolean addFilters, String... loginProps) throws Exception {
+        ServerConfiguration server = emptyConfiguration.clone();
+
+        LdapRegistry ldap = new LdapRegistry();
+
+        ldap.setRealm("LDAPRealm");
+        ldap.setHost("localhost");
+        ldap.setPort(String.valueOf(ds.getLdapPort()));
+        ldap.setBaseDN(DN);
+        ldap.setBindDN(InMemoryLDAPServer.getBindDN());
+        ldap.setBindPassword(InMemoryLDAPServer.getBindPassword());
+        ldap.setLdapType("Custom");
+        ldap.setLdapCache(new LdapCache(new AttributesCache(false, 0, 0, "0s"), new SearchResultsCache(false, 0, 0, "0s")));
+        for (String prop : loginProps) {
+            ldap.addLoginProperty(new LoginProperty(prop));
+        }
+
+        if (addFilters) {
+            ldap.setCustomFilters(new LdapFilters("(&(cn=%v)(objectclass=inetorgperson))", null, "user:postalCode", null, null));
+        }
+        server.getLdapRegistries().add(ldap);
+        updateConfigDynamically(libertyServer, server);
+    }
+
+    /**
+     * Log in the with the user's uid. No loginProperties. No userFilter defined.
+     * This is using all the defaults, including loginProperty and the generated user filter.
+     * Uid should work. Cn should fail.
+     * Additionally, we expect the user display name to be uid.
+     *
+     * @throws Exception If there was an unexpected exception.
+     */
+    @Test
+    public void testDefaultConfig() throws Exception {
+        updateLibertyServer(false);
+        assertDNsEqual("Authentication should succeed.",
+                       USER_DN, servlet.checkPassword(userUid, "password"));
+        assertNull("Authentication should not succeed.", servlet.checkPassword(userCn, "password"));
+        assertEquals(userUid, servlet.getUserDisplayName(USER_DN));
+    }
+
+    /**
+     * Log in the with the user's cn from userFilter. No loginProperties.
```

Comment:
I would say that we define a user filter which allows a user to login with their CN to clarify the test.